Project1/ItemInfo.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <array>
+
+#include "AttributeList.h"
+#include "WeaponList.h"
+
+
+enum class _COMBAT_STATS { MIGHT, HIT, AVOID, CRIT, DODGE, ENUM_END };
+using CombatStats = Enum<_COMBAT_STATS>;
+
+
+class EquipInfo {
+public:
+	const std::string _name;
+	const EnumContainer<int, CombatStats> _stats;
+
+	EquipInfo();
+	EquipInfo(std::string name);
+	EquipInfo(std::string name, std::array<int, CombatStats::size> base_stats);
+
+	int operator[] (CombatStats stat);
+};
+
+class WeaponInfo : public EquipInfo {
+public:
+	const WeaponType _type;
+	const AttribType _offensive;
+	const AttribType _defensive;
```

Comment:
Should Weapons by default be limited to one Scaling stat and one type of damage? I like the idea of weapons that normally deal hybrid damage or that scale with multiple stats.

Also, while I think specifying what stat resists it is cool, this allows weird scenarios where damage can scale negatively with enemy CON. Also, the Mindblade is supposed to reverse damage types, but that makes little sense if we allow any stat to be a resistance. Maybe just allow PHYSICAL, MAGIC, TRUE values for _defensive?

Reply:
Should we have something like the following?
const std::vector <int> _range;

Reply:
Also, I can easily see adding a weapon's hit changing based on the range used. If so, hit might work better by having a default value, but also by having a value based on range from target. Just a thought. We can always calculate accuracy drop-off elsewhere.

Reply:
1. By default, yes. The weapon gets access to the whole stat block, so if we want weapons with other formulas, we can do that.
2. This is a problem specific to that kind of skill; I don't think it's worth making the default less flexible and more complex to avoid a bit of complexity in one specific instance.
3. Yeah, I'm going to make some bitfield classes to handle this kind of thing.
4. I haven't thought about ranges yet. I think weapons with drop-off should be a specialization of this class. It's a lot more complicated.